drools-ansible-rulebook-integration-protoextractor/src/main/java/org/drools/ansible/rulebook/integration/protoextractor/prototype/CompleteExtractorPrototypeExpression.java

```diff
@@ -0,0 +1,44 @@
+package org.drools.ansible.rulebook.integration.protoextractor.prototype;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+
+import org.drools.ansible.rulebook.integration.protoextractor.ast.ExtractorNode;
+import org.drools.model.PrototypeExpression.EvaluableExpression;
+import org.drools.model.PrototypeFact;
+import org.drools.model.PrototypeVariable;
+
+public class CompleteExtractorPrototypeExpression extends ExtractorPrototypeExpression implements EvaluableExpression {
+    private final PrototypeVariable protoVar;
+
+    public CompleteExtractorPrototypeExpression(PrototypeVariable protoVar, ExtractorNode extractorNode) {
+        super(extractorNode);
+        this.protoVar = protoVar;
+    }
+
+    public PrototypeVariable getProtoVar() {
+        return protoVar;
+    }
+
+    @Override
+    public boolean hasPrototypeVariable() {
+        return true;
+    }
+
+    @Override
+    public Collection<PrototypeVariable> getPrototypeVariables() {
+        return Collections.singletonList(protoVar);
+    }
+
+    @Override
+    public Object evaluate(Map<PrototypeVariable, PrototypeFact> factsMap) {
+        PrototypeFact prototypeFact = factsMap.get(protoVar);
+        return asFunction(null).apply(prototypeFact);
```

Comment:
That's ok for now, but `asFunction` should be always called passing the `Prototype`. Please add a TODO comment for now and we will review this when we will move this entire module into Drools.

Reply:
done with https://github.com/kiegroup/drools-ansible-rulebook-integration/pull/56/commits/f6ccd3735b13703c595e46039cf76849386e098e